nym-vpnd/src/service/exit_listener.rs

```diff
@@ -18,22 +19,25 @@ impl VpnServiceExitListener {
 
     pub(super) async fn start(
         self,
-        vpn_exit_rx: futures::channel::oneshot::Receiver<nym_vpn_lib::NymVpnExitStatusMessage>,
+        vpn_exit_rx: oneshot::Receiver<nym_vpn_lib::NymVpnExitStatusMessage>,
+        listener_vpn_exit_tx: oneshot::Sender<nym_vpn_lib::NymVpnExitStatusMessage>,
     ) {
         tokio::spawn(async move {
             match vpn_exit_rx.await {
                 Ok(exit_res) => match exit_res {
                     nym_vpn_lib::NymVpnExitStatusMessage::Stopped => {
                         info!("VPN exit: stopped");
                         self.set_shared_state(VpnState::NotConnected);
+                        listener_vpn_exit_tx.send(exit_res).ok();
                     }
-                    nym_vpn_lib::NymVpnExitStatusMessage::Failed(err) => {
+                    nym_vpn_lib::NymVpnExitStatusMessage::Failed(ref err) => {
                         error!("VPN exit: fail: {err}");
                         self.set_shared_state(VpnState::ConnectionFailed(err.to_string()));
+                        listener_vpn_exit_tx.send(exit_res).ok();
                     }
                 },
                 Err(err) => {
-                    error!("exit listener fail: {err}");
+                    error!("Exit listener: {err}");
```

Comment:
I thought you were the one wanting our errors to start with lowercase : )

Reply:
Yep, for error enums, so that they can be included as substrings in e.g log entries like these :)